### PR TITLE
Deployment docker and ecm

### PR DESCRIPTION
**Pull request by benjaminbollen**
includes #157 partially, so merge after #157 is merged.

### CODE REVIEW

File: DOCKER/Dockerfile [hunk 1]
```diff
@@ -1,80 +0,0 @@
...
-
-# install glide for dependency management
-RUN	go install github.com/Masterminds/glide
-
-# set the repo and install erisdb
-ENV REPO $GOPATH/src/github.com/eris-ltd/eris-db
-WORKDIR $REPO
-COPY . $REPO
-RUN glide install
-RUN cd ./cmd/eris-db && go build -o /usr/local/bin/eris-db && cd ../..
```
Comment (silasdavis) [inline DOCKER/Dockerfile]: I don't think the final cd ../.. is needed; RUN always executes to whatever WORKDIR is set to

Comment (silasdavis) [inline DOCKER/Dockerfile]: I just commented on a deleted line.

File: tests/build_tool.sh [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+then
+  repo=`pwd`
+else
+  base=github.com/eris-ltd/eris-db
+  repo=$GOPATH/src/$base
+fi
+branch=${CIRCLE_BRANCH:=master}
+branch=${branch/-/_}
+
+release_min=$(cat $repo/version/version.go | tail -n 1 | cut -d \  -f 4 | tr -d '"')
```
Comment (silasdavis) [inline tests/build_tool.sh]: Minor but would this be less fragile if we added a $repo/version/main.go file that prints the version then we could do `$(go run $repo/version)` to get the version

**Comment by silasdavis** [comment]
LGTM

File: tests/build_tool.sh (hunk 2 above)
Comment (benjaminbollen) [inline tests/build_tool.sh]: good comment, but this should be done across the stack for consistency reasons and there is an argument for using solid bash. definitely worth raising an issue and following up

**Pull request merged**